public/static/app/perfdiscovery/PerfDiscoveryDataService.js

```diff
@@ -387,13 +388,15 @@ mciModule.factory('PerfDiscoveryDataService', function (
 
   // Makes series of HTTP calls and loads curent, baseline and history data
   // :rtype: $q.promise
-  function queryData(tasksPromise, baselineTasks) {
+  function queryData(tasksPromise, baselineTasks, expandedCurrent, expandedBaseline, expandedTrend) {
```

Comment:
This code is approaching the edge of where you would try to refactor to break it up.

Given this, it would be good to add some tests?

But that may be a large refactor given the visiblity of the functions.

Reply:
I agree. I took a stab at refactoring this so that it's more easily testable, but couldn't come up with anything that didn't require a mock HTTP backend, and I feel that's out of scope for this ticket. If you have suggestions though, I'm open to trying them.

Reply:
I kind of agree, it is too large a piece of work for this ticket.